conda/plugins/hookspec.py

```diff
@@ -301,3 +301,35 @@ def conda_post_solves():
                    action=example_post_solve,
                )
         """
+
+    @_hookspec
+    def conda_configuration_parameters(self) -> Iterable[CondaPostSolve]:
+        """
+        Register new configuration parameters
+
+        The example below defines a simple string type parameter
+
+        **Example:**
+
+        .. code-block:: python
+
+           from conda import plugins
+           from conda.common.configuration import (
+               ParameterLoader,
+               PrimitiveParameter,
+               SequenceParameter,
+           )
+
+
+           @plugins.hookimpl
+           def conda_post_solves():
+               name = "example_option"
+               description = "This is an example option"
+               loader = ParameterLoader(
+                   PrimitiveParameter(name, element_type=str), aliases=(name,)
+               )
```

Comment:
I think it might be useful to construct the loader instance later in the process and have a `loader` (a callable) _and_ a `parameter` argument (a `Parameter` instance), where the loader argument defaults to `ParameterLoader`.

Reply:
I do agree that having to provide the `ParameterLoader` object feels a little boiler-platy, but I'm not sure it's bad enough to warrant wrapping it ourselves and adding another layer of abstraction. I'm guessing that most people who end up using this will be looking at the `Context` class to see examples. It would be nice for them to be able to more-or-less copy and paste stuff from there and have it work.

If we wrapped it in another layer of abstraction, it also might be harder for authors to fully understand what's going on.

Reply:
> It would be nice for them to be able to more-or-less copy and paste stuff from there and have it work.
I don't think they'll copy-paste from conda and want an easier API, will likely not want to override the used loader at all.

E.g. this would be easier:

```
       @plugins.hookimpl
       def conda_settings():
           name = "example_option"
           description = "This is an example option"
           yield plugins.CondaSetting(
               name=name,
               description=description,
               parameter=PrimitiveParameter(name, element_type=str), aliases=(name,)
           )
```